process/smartContract/process.go

```diff
@@ -773,13 +771,15 @@ func (sc *scProcessor) isSCExecutionAfterBuiltInFunc(
 			CallType:       callType,
 			GasPrice:       vmInput.GasPrice,
 			GasProvided:    vmOutput.GasRemaining,
+			GasLocked:      vmInput.GasLocked,
 			OriginalTxHash: vmInput.OriginalTxHash,
 			CurrentTxHash:  vmInput.CurrentTxHash,
 		},
 		RecipientAddr:     vmInput.RecipientAddr,
 		Function:          function,
 		AllowInitFunction: false,
 	}
+	newVMInput.GasProvided -= vmInput.GasLocked
```

Comment:
Minor refactoring: would do this subtraction directly when setting `GasProvided` in `newVMInput` on line 773 above.

Reply:
i thought it is cleaner like this.

Reply:
done